kong/resty/dns/client.lua

```diff
@@ -702,6 +702,14 @@ local function parseAnswer(qname, qtype, answers, try_list)
   return true
 end
 
+local function close_socks(resolver)
+  for _, sock in ipairs(resolver.socks or {}) do
```

Comment:
Please move this anonymous empty table to the top-level scope of the Lua file, so that this will not create an anonymous table every time when calling it.

Reply:
Not every time, only if socks == nil, but it's a good trick in lua world